src/checker.ml

```diff
@@ -173,15 +217,26 @@ let entrypoint_touch_burrow (state, burrow_id: checker * burrow_id) : LigoOp.ope
   assert_checker_invariants state;
   (([]: LigoOp.operation list), state)
 
-let entrypoint_deposit_collateral (state, burrow_no: checker * Ligo.nat) : (LigoOp.operation list * checker) =
+let entrypoint_deposit_collateral (state, (burrow_no, tok): checker * (Ligo.nat * tok)) : (LigoOp.operation list * checker) =
   assert_checker_invariants state;
+  let _ = ensure_no_tez_given () in
   let burrow_id = (!Ligo.Tezos.sender, burrow_no) in
   let burrow = find_burrow state.burrows burrow_id in
   let _ = ensure_burrow_has_no_unclaimed_slices state.liquidation_auctions burrow_id in
-  let burrow = burrow_deposit_collateral state.parameters (tok_of_tez !Ligo.Tezos.amount) burrow in
-  let op = match (LigoOp.Tezos.get_entrypoint_opt "%burrowStoreTez" (burrow_address burrow) : unit Ligo.contract option) with
-    | Some c -> LigoOp.Tezos.unit_transaction () !Ligo.Tezos.amount c
-    | None -> (Ligo.failwith error_GetEntrypointOptFailureBurrowStoreTez : LigoOp.operation) in
+  let burrow = burrow_deposit_collateral state.parameters tok burrow in
+  let transfer =
```

Comment:
I tried that actually, but then said helper function needs to take two arguments of type `address` and we don't have named arguments, so I thought it might be a bad idea (because we could accidentally switch `from` and `to` and it would all compile just fine). Totally agree that it's verbose though, I'd also like to abstract over the pattern too :confused:

Reply:
That's too bad. In that case no worries - it's also not so bad having to explicitly specify the record. The only thing which we might still want to do is use an inlined variable for the token id to avoid accidentally writing the wrong value.